Add back --cert-dir option for kubeadm init phase certs sa

mattkelly opened the pull request:
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This fixes a regression that was introduced that got rid of the `--cert-dir` option for service account key pair generation. More info available in https://github.com/kubernetes/kubeadm/issues/1354.  

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes/kubeadm/issues/1354

**Special notes for your reviewer**:
N/A

**Does this PR introduce a user-facing change?**:
```release-note
kubeadm: add back `--cert-dir` option for `kubeadm init phase certs sa`
```

Commit:
Add back --cert-dir option for init phase certs sa

mattkelly commented:
/assign @neolit123

mattkelly commented:
Testing details:

```
_output/bin/kubeadm version
```
```
kubeadm version: &version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.0-alpha.1.236+35faaab5708d01", GitCommit:"35faaab5708d01068e8a8e829b8033d81b2313c6", GitTreeState:"clean", BuildDate:"2019-01-24T00:45:23Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"darwin/amd64"}
```

```
_output/bin/kubeadm init phase certs sa -h
```
```
Generates the private key for signing service account tokens along with its public key, and saves them into sa.key and sa.pub files. If both files already exist, kubeadm skips the generation step and existing files will be used.

Alpha Disclaimer: this command is currently alpha.

Usage:
  kubeadm init phase certs sa [flags]

Flags:
      --cert-dir string   The path where to save and store the certificates. (default "/etc/kubernetes/pki")
  -h, --help              help for sa

Global Flags:
      --log-file string   If non-empty, use this log file
      --rootfs string     [EXPERIMENTAL] The path to the 'real' host root filesystem.
      --skip-headers      If true, avoid header prefixes in the log messages
  -v, --v Level           log level for V logs
```

```
_output/bin/kubeadm init phase certs sa --cert-dir /tmp/pki
```
```
[certs] Generating "sa" key and public key
```

```
ll /tmp/pki
```
```
total 16
-rw-------  1 matt  wheel  1675 Jan 23 19:47 sa.key
-rw-------  1 matt  wheel   451 Jan 23 19:47 sa.pub
```

rosti reviewed (approved):
Thanks for your contribution @mattkelly !
/lgtm
/priority important-soon
/ok-to-test

rosti commented:
@mattkelly one tiny nit, you have double 'd' in `Read` in the release note.
/assign @fabriziopandini

neolit123 reviewed (approved):
thanks @mattkelly 
/approve
/hold

yagonobre reviewed (approved):
Thanks @mattkelly 
~can you add `kubeadm: ` prefix to release note?~

neolit123 commented:
the `kubeadm:` prefix is nice for RN is fine, yes.

mattkelly commented:
@rosti the word was actually `readd` as in `re-add`, but I changed it to `add back` as I can see why that's confusing :)

@yagonobre @neolit123 I added the `kubeadm:` prefix to the release note even though I see that comment is crossed off now. If I should remove it again, just let me know.

neolit123 commented:
/hold cancel
thanks for the update @mattkelly
